src/olympia/addons/tests/test_views.py

```diff
@@ -2534,6 +2535,89 @@ def test_filter_by_type(self):
         assert len(data['results']) == 1
         assert data['results'][0]['id'] == theme.pk
 
+    @patch('olympia.addons.models.get_featured_ids')
+    def test_filter_by_featured_no_app_no_lang(self, get_featured_ids_mock):
+        addon = addon_factory(slug='my-addon', name=u'Featured Addôn')
+        addon_factory(slug='other-addon', name=u'Other Addôn')
+        get_featured_ids_mock.return_value = [addon.pk]
+        assert addon.is_featured()
+        self.reindex(Addon)
+
+        data = self.perform_search(self.url, {'featured': 'true'})
+        assert data['count'] == 1
+        assert len(data['results']) == 1
+        assert data['results'][0]['id'] == addon.pk
+
+        # Leaving out the param value should also work.
+        data = self.perform_search(self.url + '?featured')
+        assert data['count'] == 1
+        assert len(data['results']) == 1
+        assert data['results'][0]['id'] == addon.pk
+
+    def test_filter_by_featured_app_and_langs(self):
+        fx_addon = addon_factory(slug='my-addon', name=u'Featured Addôn')
+        collection = collection_factory()
+        FeaturedCollection.objects.create(
+            collection=collection, application=amo.FIREFOX.id)
+        collection.add_addon(fx_addon)
+
+        fx_fr_addon = addon_factory(slug='my-addon', name=u'Lé Featured Addôn')
+        collection = collection_factory()
+        FeaturedCollection.objects.create(
+            collection=collection, application=amo.FIREFOX.id, locale='fr')
+        collection.add_addon(fx_fr_addon)
+
+        fn_addon = addon_factory(slug='my-addon', name=u'Featured Addôn 2 go')
+        collection = collection_factory()
+        FeaturedCollection.objects.create(
+            collection=collection, application=amo.ANDROID.id)
+        collection.add_addon(fn_addon)
+
+        fn_fr_addon = addon_factory(slug='my-addon', name=u'Lé Featured Mobil')
+        collection = collection_factory()
+        FeaturedCollection.objects.create(
+            collection=collection, application=amo.ANDROID.id, locale='fr')
+        collection.add_addon(fn_fr_addon)
+
+        addon_factory(slug='other-addon', name=u'Other Addôn')
+        self.reindex(Addon)
+
+        # searching for just Firefox should return the two Firefox collections.
+        data = self.perform_search(self.url, {'featured': 'true',
+                                              'app': 'firefox'})
+        assert data['count'] == 2 == len(data['results'])
+        ids = {data['results'][0]['id'], data['results'][1]['id']}
+        self.assertSetEqual(ids, {fx_addon.pk, fx_fr_addon.pk})
```

Comment:
Just to verify, this test is essentially testing the `null_value: ALL` querying right?

Reply:
No, it's testing when we only provide `app` so only `Q('term', **{'featured_for.application': app})` is used.  The next `assertSetEqual` is testing the `null_value: ALL` query (and the remaining ones too).